test/ruby/test_pack.rb

```diff
@@ -892,3 +892,499 @@ def test_unpack_offset
+    assert_equal("\x01\x02", Array.pack1("1000000001000000", "b*"))
+    assert_equal("", Array.pack1("1", "b0"))
+    assert_equal("\x01", Array.pack1("1", "b1"))
+    assert_equal("\x01\x00", Array.pack1("1", "b2"))
+    assert_equal("\x01\x00", Array.pack1("1", "b3"))
+    assert_equal("\x01\x00\x00", Array.pack1("1", "b4"))
+    assert_equal("\x01\x00\x00", Array.pack1("1", "b5"))
+    assert_equal("\x01\x00\x00\x00", Array.pack1("1", "b6"))
+
+    assert_equal("\xff\x00", Array.pack1("1111111100000000", "B*"))
+    assert_equal("\x01\x02", Array.pack1("0000000100000010", "B*"))
+    assert_equal("", Array.pack1("1", "B0"))
+    assert_equal("\x80", Array.pack1("1", "B1"))
+    assert_equal("\x80\x00", Array.pack1("1", "B2"))
+    assert_equal("\x80\x00", Array.pack1("1", "B3"))
+    assert_equal("\x80\x00\x00", Array.pack1("1", "B4"))
+    assert_equal("\x80\x00\x00", Array.pack1("1", "B5"))
+    assert_equal("\x80\x00\x00\x00", Array.pack1("1", "B6"))
+  end
+
+  def test_pack_unpack_hH
+    assert_equal("\x01\xfe", Array.pack1("10ef", "h*"))
+    assert_equal("", Array.pack1("10ef", "h0"))
+    assert_equal("\x01\x0e", Array.pack1("10ef", "h3"))
+    assert_equal("\x01\xfe\x0", Array.pack1("10ef", "h5"))
+    assert_equal("\xff\x0f", Array.pack1("fff", "h3"))
+    assert_equal("\xff\x0f", Array.pack1("fff", "h4"))
+    assert_equal("\xff\x0f\0", Array.pack1("fff", "h5"))
+    assert_equal("\xff\x0f\0", Array.pack1("fff", "h6"))
+    assert_equal("\xff\x0f\0\0", Array.pack1("fff", "h7"))
+    assert_equal("\xff\x0f\0\0", Array.pack1("fff", "h8"))
+
+    assert_equal("\x10\xef", Array.pack1("10ef", "H*"))
+    assert_equal("", Array.pack1("10ef", "H0"))
+    assert_equal("\x10\xe0", Array.pack1("10ef", "H3"))
+    assert_equal("\x10\xef\x0", Array.pack1("10ef", "H5"))
+    assert_equal("\xff\xf0", Array.pack1("fff", "H3"))
+    assert_equal("\xff\xf0", Array.pack1("fff", "H4"))
+    assert_equal("\xff\xf0\0", Array.pack1("fff", "H5"))
+    assert_equal("\xff\xf0\0", Array.pack1("fff", "H6"))
+    assert_equal("\xff\xf0\0\0", Array.pack1("fff", "H7"))
+    assert_equal("\xff\xf0\0\0", Array.pack1("fff", "H8"))
+  end
+
+  def test_pack_unpack_sS
+    assert_equal(2, Array.pack1(1, "s").bytesize)
+    assert_equal(2, Array.pack1(1, "S").bytesize)
+    assert_operator(2, :<=, Array.pack1(1, "s!").bytesize)
+    assert_operator(2, :<=, Array.pack1(1, "S!").bytesize)
+  end
+
+  def test_pack_unpack_iI
+    assert_operator(4, :<=, Array.pack1(1, "i").bytesize)
+    assert_operator(4, :<=, Array.pack1(1, "I").bytesize)
+    assert_operator(4, :<=, Array.pack1(1, "i!").bytesize)
+    assert_operator(4, :<=, Array.pack1(1, "I!").bytesize)
+  end
+
+  def test_pack_unpack_lL
+    assert_equal(4, Array.pack1(1, "l").bytesize)
+    assert_equal(4, Array.pack1(1, "L").bytesize)
+    assert_operator(4, :<=, Array.pack1(1, "l!").bytesize)
+    assert_operator(4, :<=, Array.pack1(1, "L!").bytesize)
+  end
+
+  require 'rbconfig'
```

Comment:
I think this and the original line in `TestPack` should be at the top of this file.